### PR TITLE
Feat: Added Modal component

### PR DESCRIPTION
**Pull request by anurag-kkumarr**
- Added simple dialog box
- Added dark mode dialog box
- created a demo dialog box for preview

### CODE REVIEW

**Review by Asuraking1n — COMMENTED** [review]
use Some paddings
also cancel buttons are not working in model 
and model navigation is absent on isbar

File: components/CSS/modal.css [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+.basic-modal {
+    border: 1px solid black;
+    background-color: white;
+    width: 400px;
+    height: 200px;
+    padding: 2rem;
+    position: relative;
+}
+.dark-md {
+    background-color: #121212;
```
Comment (sakshi006) [inline components/CSS/modal.css]: Can use CSS variable here as well

File: components/CSS/modal.css [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    font-weight: bold;
+    border: 1px solid white;
+    border-radius: 5px;
+    cursor: pointer;
+   background-color: var(--color-primary);
+}
+.btn-container {
+    width: 100%;
+    padding: 1rem;
+    border: 1px solid black;
```
Comment (sakshi006) [inline components/CSS/modal.css]: Can use CSS variables

File: components/CSS/modal.css [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    bottom: 0.5rem;
+    right: 0.5rem;
+}
+.demo-btn {
+    width: 4rem;
+    padding: 1rem;
+    color: white;
+    font-size: 1rem;
+    font-weight: bold;
+    border: 1px solid white;
```
Comment (sakshi006) [inline components/CSS/modal.css]: Can use CSS variables

**Review by sakshi006 — APPROVED** [review]
Close button of modal is not working
Also, for the sidebar, the transitions on hover are not smooth

**Pull request merged**